modules/scip/9.2.0.bcr.1/patches/add_build_file.patch

```diff
@@ -0,0 +1,161 @@
+Diff --git a/BUILD.bazel b/BUILD.bazel
+new file mode 100644
+index 0000000000..0c3f1853b4
+--- /dev/null
++++ b/BUILD.bazel
+@@ -0,0 +1,155 @@
++# Copyright 2010-2024 Google LLC
++# Licensed under the Apache License, Version 2.0 (the "License");
++# you may not use this file except in compliance with the License.
++# You may obtain a copy of the License at
++#
++#     http://www.apache.org/licenses/LICENSE-2.0
++#
++# Unless required by applicable law or agreed to in writing, software
++# distributed under the License is distributed on an "AS IS" BASIS,
++# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
++# See the License for the specific language governing permissions and
++# limitations under the License.
++
++exports_files(
++    ["src/lpi/lpi_glop.cpp"],
++)
++
++config_setting(
++    name = "on_linux",
++    constraint_values = [
++        "@platforms//os:linux",
++    ],
++)
++
++config_setting(
++    name = "on_macos",
++    constraint_values = [
++        "@platforms//os:macos",
++    ],
++)
++
++config_setting(
++    name = "on_windows",
++    constraint_values = [
++        "@platforms//os:windows",
++    ],
++)
++
++PLATFORM_FLAGS = select({
++    "on_linux": [
++        "-Wunknown-pragmas",
++        "-fexceptions",
++        "-DSYM=bliss",
++    ],
++    "on_macos": [
++        "-Wunknown-pragmas",
++        "-fexceptions",
++        "-DSYM=bliss",
++    ],
++    "on_windows": [
++        "/DSYM=none",
++        "/DSCIP_NO_SIGACTION",
++        "/DSCIP_NO_STRTOK_R",
++    ],
++    "//conditions:default": [],
++})
++
++PLATFORM_DEPS = select({
++    "on_linux": ["@bliss"],
++    "on_macos": ["@bliss"],
++    "on_windows": [],
++    "//conditions:default": [],
++})
++
++BLISS_FILE = select({
++    "on_linux": ["src/symmetry/compute_symmetry_bliss.cpp"],
++    "on_macos": ["src/symmetry/compute_symmetry_bliss.cpp"],
++    "on_windows": ["src/symmetry/compute_symmetry_none.cpp"],
++    "//conditions:default": ["src/symmetry/compute_symmetry_none.cpp"],
++})
++
++BASE_SRCS = glob(
++    [
++        "src/*/*.c",
++    ],
++    exclude = [
++        "src/lpi/lpi_*.c",
++        "src/nauty/*",
++        "src/scip/exprinterpret_*.c",
++        "src/scip/nlpi_filtersqp.c",
++        "src/scip/nlpi_worhp.c",
++        "src/scip/*_xyz.c",
++        "src/scip/sorttpl.c",
++        "src/symmetry/compute_symmetry_*.cpp",
++        "src/symmetry/*nauty*",
++        "src/tpi/tpi_*.c",
++    ],
++)
++
++BASE_HDRS = glob(
++    [
++        "src/*/*.h",
++        "src/*/*.hpp",
++        "src/scip/githash.c",
```

Comment:
Nit: Should we add the define for the Git hash directly in the BUILD file instead of patching it in as separate file? I assume people are more likely to adapt the BUILD file for new versions than reading through the other patches when there are no conflicts.

Reply:
Done